sdk/src/main/java/com/mapbox/maps/renderer/RendererError.kt

```diff
@@ -0,0 +1,28 @@
+package com.mapbox.maps.renderer
+
+import com.mapbox.maps.MapView
+import javax.microedition.khronos.egl.EGL10
+
+/**
+ * Typed enum wrapping up all the EGL error codes represented as ints.
+ */
+sealed class RendererError(val eglErrorCode: Int) {
+  /**
+   * Error code returned when no matching EGL config was found meaning that map rendering is not supported on given device.
+   * This could be also checked with static method [MapView.isRenderingSupported] prior to creating the [MapView].
+   */
+  object NoValidEglConfigFound : RendererError(0)
+  /**
+   * Represents out of memory caused by [EGL10.EGL_BAD_ALLOC].
+   *
+   * Could be thrown when creating context, surface or display and system is running low on resources.
+   * Best action will be try to release Android resources and re-create the [MapView].
+   */
+  object OutOfMemory : RendererError(EGL10.EGL_BAD_ALLOC)
```

Comment:
are these two cases the only special errors? would we add something later as well? to avoid breaking change, we could probably use 
```
class RendererError(val eglErrorCode: Int) {
  companion object {
    object NoValidEglConfigFound : RendererError(0)
    object OutOfMemory : RendererError(EGL10.EGL_BAD_ALLOC)
  }
}
```
so that we can add more objects without breaking changes later..

Reply:
done!